x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/pivot/AggregationResultUtils.java

```diff
@@ -211,17 +215,44 @@ public Object value(Aggregation agg, String fieldType) {
         }
     }
 
+    static class SingleBucketAggExtractor implements AggValueExtractor {
+        @Override
+        public Object value(Aggregation agg, String fieldType, Map<String, String> fieldTypeMap, String lookupFieldPrefix) {
+            SingleBucketAggregation aggregation = (SingleBucketAggregation) agg;
+
+            if (aggregation.getAggregations().iterator().hasNext() == false) {
+                return aggregation.getDocCount();
+            }
+
+            HashMap<String, Object> nested = new HashMap<>();
+            for (Aggregation subAgg : aggregation.getAggregations()) {
+                String subLookupFieldPrefix = lookupFieldPrefix.isEmpty() ? agg.getName() : lookupFieldPrefix + "." + agg.getName();
+                nested.put(
+                    subAgg.getName(),
+                    getExtractor(subAgg).value(
+                        subAgg,
+                        fieldTypeMap.get(subLookupFieldPrefix + "." + subAgg.getName()),
```

Comment:
It strikes me that supplying `fieldType` is no longer necessary. 

Anything that desires to know `fieldType` could look it up via its own agg name combined with the prefix (if not empty).